### PR TITLE
Issue/fix dataraces in jail tests#457

### PR DESCRIPTION
**Pull request by b00ris**
Fix dataraces related to jail tests

Solves #457

### CODE REVIEW

File: e2e/jail/jail_test.go [hunk 1]
```diff
@@ -146,10 +146,11 @@ func (s *JailTestSuite) TestEventSignal() {
 	opCompletedSuccessfully := make(chan struct{}, 1)
 
 	// replace transaction notification handler
+	defer signal.ResetDefaultNodeNotificationHandler()
 	signal.SetDefaultNodeNotificationHandler(func(jsonEvent string) {
 		var envelope signal.Envelope
-		err = json.Unmarshal([]byte(jsonEvent), &envelope)
-		s.NoError(err)
+		er := json.Unmarshal([]byte(jsonEvent), &envelope)
```
Comment (influx6) [inline e2e/jail/jail_test.go]: where the use of `err` part of the data race causes?

Comment (adambabik) [reply e2e/jail/jail_test.go]: Yeah, why do we change that? It should be affecting anything and `er` looks like a typo.

Comment (b00ris) [reply e2e/jail/jail_test.go]: @influx6 There is the datarace at writing to err var). 
err was defined upper. Than signal is called from jail.

@adambabik I don't see any reasons to affecting anything. SetDefaultNodeNotificationHandler is used only in tests.

Comment (adambabik) [reply e2e/jail/jail_test.go]: I see. Ok, let's at least change it to something meaningful like `unmarshalErr` or something.

Comment (b00ris) [reply e2e/jail/jail_test.go]: @adambabik just renamed.

**Review by JekaMas — APPROVED** [review]

**Review by adambabik — APPROVED** [review]
Thanks, it looks perfect now!

**Comment by adambabik** [comment]
@b00ris but something is wrong with e2e tests though... They should pass.

**Review by tiabc — APPROVED** [review]

**Comment by tiabc** [comment]
Please, fix CI before merging.
Also, branch name is invalid, it should be `bugfix/dataraces-in-jail-tests-457` due to our naming conventions.

**Pull request merged**